src/test/scala/com/microsoft/hyperspace/index/IndexLogEntryTest.scala

```diff
@@ -139,12 +178,176 @@ class IndexLogEntryTest extends SparkFunSuite {
+    val content = Content(Directory("file:/", subDirs =
+      Seq(
+        Directory("a",
+          files = Seq(FileInfo("f1", 0, 0), FileInfo("f2", 0, 0)),
+          subDirs = Seq(
+            Directory("b",
+              files = Seq(FileInfo("f3", 0, 0), FileInfo("f4", 0, 0)))))
+      )))
+
+    val expected =
+      Seq("file:/a/f1", "file:/a/f2", "file:/a/b/f3", "file:/a/b/f4").map(new Path(_)).toSet
+    val actual = content.files.toSet
+    assert(actual.equals(expected))
+  }
+
+  test("Content.fromDirectory api creates the correct Content object.") {
+    val nestedDirPath = toPath(nestedDir)
+
+    val expected = {
+      val fileInfos = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+      val nestedDirDirectory = Directory("nested", fileInfos)
+      val rootDirectory = createDirectory(nestedDirPath, nestedDirDirectory)
+      Content(rootDirectory, NoOpFingerprint())
+    }
+
+    val actual = Content.fromDirectory(nestedDirPath)
+    assert(contentEquals(actual, expected))
+  }
+
+  test("Content.fromLeafFiles api creates the correct Content object.") {
+    val nestedDirPath = toPath(nestedDir)
+
+    val expected = {
+      val fileInfos = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+      val nestedDirDirectory = Directory("nested", fileInfos)
+      val rootDirectory = createDirectory(nestedDirPath, nestedDirDirectory)
+      Content(rootDirectory, NoOpFingerprint())
+    }
+
+    val actual = Content.fromDirectory(nestedDirPath)
+    assert(contentEquals(actual, expected))
+  }
+
+  test("Directory.fromDirectory api creates the correct Directory object.") {
+    val nestedDirPath = toPath(nestedDir)
+
+    val expected = {
+      val fileInfos = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+      val nestedDirDirectory = Directory("nested", fileInfos)
+      createDirectory(nestedDirPath, nestedDirDirectory)
+    }
+
+    val actual = Directory.fromDirectory(nestedDirPath)
+    assert(directoryEquals(actual, expected))
+  }
+
+  test("Directory.fromDirectory api creates the correct Directory objects, " +
+    "recursively listing all leaf files.") {
+    val testDirPath = toPath(testDir)
+
+    val testDirLeafFiles = Seq(f1, f2).map(toFileStatus).map(FileInfo(_))
+    val nestedDirLeafFiles = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+    val testDirDirectory = Directory(name = "testDir",
+      files = testDirLeafFiles,
+      subDirs = Seq(Directory(name = "nested", files = nestedDirLeafFiles)))
+    val expected = createDirectory(testDirPath, testDirDirectory)
+
+    val actual = Directory.fromDirectory(testDirPath)
+
+    assert(directoryEquals(actual, expected))
+  }
+
+  test("Directory.fromLeafFiles api creates the correct Directory object.") {
+    val testDirPath = toPath(testDir)
+
+    val testDirLeafFiles = Seq(f1, f2).map(toFileStatus).map(FileInfo(_))
+    val nestedDirLeafFiles = Seq(f3, f4).map(toFileStatus).map(FileInfo(_))
+    val testDirDirectory = Directory(name = "testDir",
```

Comment:
Can you also add tests where files are in the same level, but different dirs:
```
/tmp/a/f1.txt
/tmp/b/f1.txt
```

and a test with a gap in the directory (no files)

```
/tmp/a/f1.txt
/tmp/b/c/f1.txt
```

and multiple subdirs?
```
/tmp/a/f1.txt
/tmp/a/b/f1.txt
/tmp/a/c/f1.txt
```

Reply:
thanks @imback82 , added the suggested tests, although I separated out these tests from other tests in terms of setup, for clarity.